streams/src/main/java/org/apache/kafka/streams/processor/internals/StreamThread.java

```diff
@@ -768,9 +770,30 @@ void runOnce() {
 
     private void resetInvalidOffsets(final InvalidOffsetException e) {
         final Set<TopicPartition> partitions = e.partitions();
+        final Set<TopicPartition> notReset = resetOffsets(partitions);
+        if (!notReset.isEmpty()) {
+            final String notResetString =
+                notReset.stream()
+                        .map(tp -> "topic " + tp.topic() + "(partition " + tp.partition() + ")")
+                        .collect(Collectors.joining(","));
```

Comment:
Nit: As reset policy is set on a per topic basis, it's sufficient to list the topic names -- it does not add value if we list the partitions, because all assigned partitions would be affected anyway.

Reply:
Interesting; good point. I just copied over the original message. Do you think I should drop the partition?

Reply:
I think it would be good. Seems the original message should not have contained the partition to begin with. Even if for the original one, it's less severe because we only consider a single topic-partition anyway. For this case, we get multiple topic-partitions and the error message might become quite large? Also, if multiple task of the same sub-topology are assigned, the same topic name might appear multiple times? Might be best to only add a `Set` or topic names to the error messaage?

Reply:
Ok, I've done this.